### DISCUSSION TITLE
why was the java version specified in compileOptions removed from the template?

### DISCUSSION THREAD

**Discussion by Adnan-Bacic** [react-native-community/discussions-and-proposals#768] [Feedback]
i have the version of node in the .nvmrc file

the version of yarn in the .yarnrc.yml file

the version of ruby and cocoapods on the gemfile file

but nowhere is the project version of java specified. i know react native 0.73 will upgrade from 11 to 17 but this is not reflected anywhere in the code. i would have to run `java --version` to see the version. i think it would be very handy if this value was embedded into the project.

i know previous version of react native had this code:

    compileOptions {
            sourceCompatibility JavaVersion.VERSION_1_8
            targetCompatibility JavaVersion.VERSION_1_8
        }

on 0.72 that version that would be

        compileOptions {
            sourceCompatibility JavaVersion.VERSION_11
            targetCompatibility JavaVersion.VERSION_11
        }

and i guess it would be JavaVersion.VERSION_17 on 0.73.

i only remember this because when they updated from 8 to 11 this was already removed and someone suggested adding it here: https://github.com/facebook/react-native/issues/33731 to fix a problem someone was encountering

im just curious as to why this was removed from the template. it seems like it adds a lot of value having the version specific, especially since certain versions of react native rely on a specific java version.

**Comment by cortinico** [comment]
> im just curious as to why this was removed from the template. it seems like it adds a lot of value having the version specific, especially since certain versions of react native rely on a specific java version.

It has been removed because React Native Gradle Plugin sets the Java version for all the modules in your project so there is no need to specify it in your project. That's one field less for you to configure so one thing less that can go wrong when building things
